WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -362,6 +366,48 @@ - (ReaderTopic *)siteTopicForPost:(ReaderPost *)post
     return topic;
 }
 
+- (void)siteTopicForSiteWithID:(NSNumber *)siteID
+                       success:(void (^)(NSManagedObjectID *objectID, BOOL isFollowing))success
+                       failure:(void (^)(NSError *error))failure
+{
+    ReaderTopicServiceRemote *remoteService = [[ReaderTopicServiceRemote alloc] initWithApi:[self apiForRequest]];
+    [remoteService fetchSiteInfoForSiteWithID:siteID success:^(RemoteReaderSiteInfo *siteInfo) {
+        if (!success) {
+            return;
+        }
+        NSEntityDescription *entity = [NSEntityDescription entityForName:NSStringFromClass([ReaderTopic class])
+                                                  inManagedObjectContext:self.managedObjectContext];
+        ReaderTopic *topic = [[ReaderTopic alloc] initWithEntity:entity
+                     insertIntoManagedObjectContext:self.managedObjectContext];
```

Comment:
Thanks Eric!. Notifications will probably be instantiating just the `ReaderBrowseSiteViewController`, and shouldn't directly use this service.

Main context sounds good to me, if we were to measure the success block, it'd probably be in the milliseconds range.